homeassistant/components/octoprint/camera.py

```diff
@@ -0,0 +1,63 @@
+"""Support for monitoring OctoPrint binary sensors."""
+from __future__ import annotations
+
+from pyoctoprintapi import OctoprintClient
+
+from homeassistant.components.mjpeg.camera import MjpegCamera
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import OctoprintDataUpdateCoordinator
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the available OctoPrint camera."""
+    coordinator: OctoprintDataUpdateCoordinator = hass.data[DOMAIN][
+        config_entry.entry_id
+    ]["coordinator"]
+    client: OctoprintClient = hass.data[DOMAIN][config_entry.entry_id]["client"]
+    device_id = config_entry.unique_id
+
+    assert device_id is not None
```

Comment:
Instead of throwing an `AssertionError` that the user can't do anything about, how about just returning if needed?

```suggestion
if not config_entry.unique_id:
    return
```

Reply:
The other platforms in the integration are currently doing the same thing as here, so I am a little hesitant to introduce different behavior here. 
Is there a valid case where the unique_id would be null and the code gets to this point?

Reply:
As other platforms do it, I think it is OK for now. In general, asserts should be avoided (it generally is working around a typing issue that should be addressed instead).